### PR TITLE
SegmentTree Template

### PR DESCRIPTION
**Pull request by cancanneedcoding**

### CODE REVIEW

File: DataStructure/HLD.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        dep.resize(n);
+        parent.resize(n);
+        in.resize(n);
+        out.resize(n);
+        seq.resize(n);
+        cur = 0;
+        adj.assign(n, {});
+    }
+
+    // 在节点u和v之间添加一条无向边。
```
Comment (QuakeWang) [inline DataStructure/HLD.cpp]:
```suggestion
    // 在节点 u 和 v 之间添加一条无向边。
```

format

File: DataStructure/HLD.cpp [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            dfs1(v);
+            siz[u] += siz[v];
+            if (siz[v] > siz[adj[u][0]])
+            {
+                std::swap(v, adj[u][0]);
+            }
+        }
+    }
+
+    // 第二遍深度优先搜索，分配每个节点的进时间戳in[u]，建立重链，并计算每个节点的出时间戳out[u]。
```
Comment (QuakeWang) [inline DataStructure/HLD.cpp]:
```suggestion
    // 第二遍深度优先搜索，分配每个节点的进时间戳 in[u]，建立重链，并计算每个节点的出时间戳 out[u]。
```
format

File: DataStructure/HLD.cpp [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        seq[in[u]] = u;
+        for (auto v : adj[u])
+        {
+            top[v] = v == adj[u][0] ? top[u] : v;
+            dfs2(v);
+        }
+        out[u] = cur;
+    }
+
+    // 计算节点u和v的最近公共祖先（LCA）
```
Comment (QuakeWang) [inline DataStructure/HLD.cpp]:
```suggestion
    // 计算节点 u 和 v 的最近公共祖先（LCA）
```

File: DataStructure/HLD.cpp [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            }
+            else
+            {
+                v = parent[top[v]];
+            }
+        }
+        return dep[u] < dep[v] ? u : v;
+    }
+
+    // 计算节点u和v之间的距离。
```
Comment (QuakeWang) [inline DataStructure/HLD.cpp]:
```suggestion
    // 计算节点 u 和 v 之间的距离。
```

File: DataStructure/HLD.cpp [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+        while (dep[top[u]] > d)
+        {
+            u = parent[top[u]];
+        }
+
+        return seq[in[u] - dep[u] + d];
+    }
+
+    // 判断节点u是否是节点v的祖先
```
Comment (QuakeWang) [inline DataStructure/HLD.cpp]:
```suggestion
    // 判断节点 u 是否是节点 v 的祖先
```

**Review by QuakeWang — COMMENTED** [review]
@cancanneedcoding For a better reading experience, you should standardise the formatting of all documents.

ref: [chinese-copywriting-guidelines](https://github.com/sparanoid/chinese-copywriting-guidelines)

**Comment by cancanneedcoding** [comment]
1111

**Pull request merged**